### PR TITLE
MAGECLOUD-3502: Rewrite docker integration tests to functional tests using Codeception

### PR DESCRIPTION
**Pull request by BaDos**
### Description

Docker integration tests were rewritten using Codeception.
Also, these tests were renamed functional.

### Fixed Issues (if relevant)

https://magento2.atlassian.net/browse/MAGECLOUD-3502

### Manual testing scenarios
N/A

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -49,7 +51,8 @@
     "psr-4": {
       "Magento\\MagentoCloud\\Test\\": "tests/static/",
       "Magento\\MagentoCloud\\Test\\Integration\\": "tests/integration/",
-      "Magento\\MagentoCloud\\Test\\DockerIntegration\\": "tests/docker-integration/"
+      "Magento\\MagentoCloud\\Test\\DockerIntegration\\": "tests/docker-integration/",
+      "Magento\\MagentoCloud\\Test\\DockerFunctional\\": "tests/docker-functional/"
```
Comment (shiftedreality) [inline composer.json]: rename it to `functional`

File: tests/docker-functional/_support/_generated/.gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+*
+!.gitignore
```
Comment (shiftedreality) [inline tests/docker-functional/_support/_generated/.gitignore]: Add newline

File: tests/travis/script.sh [hunk 3]
```diff
@@ -49,4 +49,13 @@ case $TEST_SUITE in
                 ;;
         esac
         ;;
+    docker-functional)
```
Comment (shiftedreality) [inline tests/travis/script.sh]: also rename to fucntional

File: composer.json (hunk 1 above)
Comment (BaDos) [reply composer.json]: done

**Review by shiftedreality — APPROVED** [review]

**Comment by shiftedreality** [comment]
Seems nice.
@bbatsche please take a look as well

File: tests/travis/script.sh (hunk 3 above)
Comment (BaDos) [reply tests/travis/script.sh]: It was renamed.
Maybe you saw an outdated commit.

File: tests/docker-functional/_support/_generated/.gitignore (hunk 2 above)
Comment (BaDos) [reply tests/docker-functional/_support/_generated/.gitignore]: Done

File: tests/functional/Codeception/Docker.php [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+     *
+     * @param string|null $version
+     * @return bool
+     * @throws \Robo\Exception\TaskException
+     */
+    public function cloneTemplate(string $version = null): bool
+    {
+        $gitTask = $this->taskGitStack()
+            ->exec('git init')
+            ->exec('git remote -vvv')
```
Comment (oshmyheliuk) [inline tests/functional/Codeception/Docker.php]: why do you need to run `git remote -vvv`?  Looks like this command is redundant.

Comment (BaDos) [reply tests/functional/Codeception/Docker.php]: Yes, it should be removed.
I used it during development.
Thanks!

File: tests/functional/Codeception/Docker.php [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+
+    /**
+     * Cleans directories
+     *
+     * @param string|array $path
+     * @param string $container
+     * @return bool
+     * @throws \Robo\Exception\TaskException
+     */
+    public function cleanDirectories($path, string $container = self::BUILD_CONTAINER): bool
```
Comment (oshmyheliuk) [inline tests/functional/Codeception/Docker.php:153]: According to the method name, it should accept some array of paths not a single path.
It's not a good approach to build logic on the provided argument type.

Looks like it's better to create two methods `cleanDirectory` and `cleanDirectories` or accept only arrays as an argument.

Comment (BaDos) [reply tests/functional/Codeception/Docker.php:153]: Good point.
I'll think about it.

**Comment by bbatsche** [comment]
Looks like some really good work, but why bring in Codeception? Couldn't most of this have been done with PHPUnit still?

**Review by oshmyheliuk — APPROVED** [review]

**Comment by shiftedreality** [comment]
@bbatsche we're unifying our tools because at some point I'm started to invent the same concepts

**Comment by BaDos** [comment]
So... Waiting for green tests and merge it :)

**Pull request merged**